### ISSUE TITLE
Merge SC1001, SC1012

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#1258]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1001, SC1012
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
echo 'tr '\0' '\n' | sort'

```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
 
Line 3:
echo 'tr '\0' '\n' | sort'
          ^-- SC1001: This \0 will be a regular '0' in this context.
          ^-- SC2026: This word is outside of quotes. Did you intend to 'nest '"'single quotes'"' instead'? 
               ^-- SC1012: \n is just literal 'n' here. For line feed, use a quoted, literal line feed instead.
               ^-- SC2026: This word is outside of quotes. Did you intend to 'nest '"'single quotes'"' instead'? 
```

#### Here's what I wanted or expected to see:

SC1001 and SC1012 merged. SC1001 covers the whole category, and SC1012 wiki page does not exist.

**Comment by koalaman** [comment]
There's definitely an argument for making these the same. They're currently differentiated by level, with SC1012 being warnings that very definitely should be escaped, while SC1001 are info messages for sequences with less certain intent. For now I've left them separate, but added the missing wiki page.

Thanks!
